apps/hash-graph/libs/api/src/rpc/mod.rs

```diff
@@ -13,3 +16,32 @@ mod role {
     pub(crate) type Server = harpc_service::role::Server<Session<Account>>;
     pub(crate) type Client<Svc, C> = harpc_service::role::Client<Connection<Svc, C>>;
 }
+
+#[derive(Debug, Copy, Clone)]
+pub enum GraphSubsystemId {
+    Echo,
+    Authentication,
+    Account,
+}
+
+impl SubsystemIdentifier for GraphSubsystemId {
+    fn from_id(id: SubsystemId) -> Option<Self>
+    where
+        Self: Sized,
+    {
+        match id.value() {
+            0x00 => Some(Self::Echo),
+            0x01 => Some(Self::Authentication),
+            0x02 => Some(Self::Account),
+            _ => None,
+        }
+    }
+
+    fn into_id(self) -> SubsystemId {
+        match self {
+            Self::Echo => SubsystemId::new(0x00),
+            Self::Authentication => SubsystemId::new(0x01),
+            Self::Account => SubsystemId::new(0x02),
+        }
+    }
+}
```

Comment:
Do we want to use `From` instead?

Reply:
We'd need to use `TryFrom`, but there isn't really an error here. Simply that it doesn't match. Otherwise we could say: `SubsystemIdentifier: Into<SubsystemId>`. I wanted to keep this separate tho to make it clear that it's a specific operation with intent. In the future (once it lands as a non-incomplete feature) I'd like to make these actually const as a requirement.

Reply:
Makes sense!